### ISSUE TITLE
Fix for warning on float 0.0

### ISSUE THREAD

**Issue by kivra-pauoli** [tomas-abrahamsson/gpb#230]
👋 

A recent [change to OTP](https://github.com/erlang/otp/pull/7521) includes a warning on matching on float 0.0, in the form of

```
matching on the float 0.0 will no longer also match -0.0 in OTP 27. If you specifically intend to match 0.0 alone, write +0.0 instead.
```

In anticipation of OTP 27, would you think it possible `gpb` was updated to generate e.g.

```erlang
if TrF3 =:= +0.0 -> B2;
```

instead of

```erlang
if TrF3 =:= 0.0 -> B2;
```

?

I'd pull request if I knew all the details behind how that's generated and if we actually want restrictions around +0.0 or -0.0.

Thanks.

**Comment by kivra-pauoli** [comment]
@tomas-abrahamsson, I don't know how the `gpb` releases you update at `rebar3_gpb_plugin` get Hex-released. Should I request a release in that repo. after this is taken care of, or do you usually do it? (at the moment, Hex says 2.22.1, while GitHub tag says 2.22.7).

**Comment by kivra-pauoli** [comment]
In the meantime, on the project including the generated files one can do this in `rebar.config` to avoid the warning:

```erlang
{erl_opts, [nowarn_match_float_zero]}.
```

**Comment by tomas-abrahamsson** [comment]
Yes, I noticed the warn about floats, both in the generated code and in gpb's own eunit tests, and have started to look into it. Good point about `nowarn_match_float_zero` meanwhile.

For rebar3_gpb_plugin, the workflow is that I submit a PR to that repo about the uplift, and I should think @lrascao cuts a release to hex every once in a while, but I don't know the cadence. So, yes, the best is probably to to request a release in that repo after this has been taken care of.

**Comment by kivra-pauoli** [comment]
Thank you for taking a look at this.

**Comment by tomas-abrahamsson** [comment]
The 0.0 issue should be fixed now in 4.20.0 which is just released, so I'm closing this.
